api/server/server_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package server
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/snow"
+)
+
+func TestRejectMiddleware(t *testing.T) {
+	type test struct {
+		name            string
+		handlerFunc     func(*require.Assertions) http.Handler
+		contextFunc     func() *snow.ConsensusContext
+		expectedErrCode int
+	}
+
+	tests := []test{
+		{
+			name: "chain is bootstrapping",
+			handlerFunc: func(require *require.Assertions) http.Handler {
+				return http.HandlerFunc(func(_ http.ResponseWriter, _ *http.Request) {
```

Comment:
(No action required) Maybe define only the anonymous inner func and leave the boilerplate to the actual test so it isn't duplicated?

Reply:
I don't think that's possible because the inner function needs to get the reference to `require` somehow.